test/e2e/README.md

```diff
@@ -0,0 +1,156 @@
+type PluginCmdOps interface {
+    PluginBasicOps
+    PluginSourceOps
+}
+```
+
+To perform cluster specific operations:
+
+```go
+// ClusterOps has helper operations to perform on cluster
+type ClusterOps interface {
+    CreateCluster(name string, args []string) (output string, err error)
+    DeleteCluster(name string, args []string) (output string, err error)
+    ClusterStatus(name string, args []string) (output string, err error)
+}
+
+// KindCluster performs k8s KIND cluster operations
+type KindCluster interface {
+    ClusterOps
+}
+```
+
+To perform tanzu config command and CLI lifecycle operations:
+
+```go
+// ConfigLifecycleOps performs "tanzu config" command operations
+type ConfigLifecycleOps interface {
+    ConfigSetFeatureFlag(path, value string) error
+    ConfigGetFeatureFlag(path string) (string, error)
+    ConfigUnsetFeature(path string) error
+    ConfigInit() error
+    ConfigServerList() error
+    ConfigServerDelete(serverName string) error
+    DeleteCLIConfigurationFiles() error
+    IsCLIConfigurationFilesExists() bool
+}
+
+// CliOps performs basic cli operations
+type CliOps interface {
+    CliInit() error
+    CliVersion() (string, error)
+    InstallCLI(version string) error
+    UninstallCLI(version string) error
+}
+```
+
+## Use cases covered in E2E tests
+
+### End user operations
+
+E2E tests are written to validate all CLI Core functionalities from the end-user perspective. They cover all CLI Core commands lifecycle operations. Below is the list of CLI Core commands or use cases covered by the E2E tests:
+
+- CLI lifecycle operations, like build and install the CLI in all possible ways and on all platforms (TODO)
+- CLI Config command lifecycle operations, like init, get, set, unset and server related operations (In-progress)
+- CLI Plugin command lifecycle operations, like install, upgrade, list, delete and discovery source operations (In-progress)
+- CLI Context command lifecycle operations, like create, get, list, delete and use context operations, including target (k8s and TMC) specific use cases (In-progress)
+- Other CLI commands lifecycle operations, like update, version, completion and init (TODO)
+
+### plugin compatibility/coexistence tests
+
+The E2E framework tests plugin compatibility by using the current version of the CLI Core and performing basic plugin operations (add/list/delete plugin, and invoke plugin basic commands - info, help) on plugins built using older versions of the CLI Plugin Runtime library. This ensures that the current CLI supports all older plugins and all plugins can coexists.
+
+## How and when E2E tests are executed
+
+E2E tests are executed as Github runner CI pipelines. The CLI Core E2E test CI pipelines will be executed for every PR created on the CLI Core repository. The E2E tests are organized as list of CLI Commands/use cases and plugin compatibility tests in Github CI pipeline, it does shows the rest results also.
```

Comment:
```suggestion
E2E tests are executed as Github runner CI pipelines. The CLI Core E2E test CI pipelines will be executed for every PR created on the CLI Core repository. The E2E tests are organized as a list of CLI commands/use cases and plugin compatibility tests in Github CI pipelines, it does shows the rest results also.
```

I don't understand what you mean by "it does shows the rest results also."?  Are we talking about the REST protocol?

Reply:
I have updated it, its `test cases ` not `rest`.